### ISSUE TITLE
Sync notes through git

### ISSUE THREAD

**Issue by saket** [saket/press#52]
I've spent a lot of time thinking about this and I think the best way of offering syncing is by not maintaining my own syncing backend. Syncing is a hard problem. The plan is to piggyback on git for syncing notes to a remote git repository. It'll have some great benefits by letting users,
- own their notes
- use a different app on platforms unsupported by Press 
- and even move on to a different markdown app if they decide Press is crap

I understand that by doing this I'm effectively restricting Press to nerds but I'm okay with that. Press will let users login using their github/gitlab/bitbucket account for auto-adding an SSH key to their account.

**Comment by saket** [comment]
I think I'm nearly done for a first cut. Just gotta figure out how to correctly resolve merge conflicts with file deletions because [Jgit](https://www.eclipse.org/jgit/) has incorrect implementation of merge strategies.

**Comment by okmanideep** [comment]
@saket Curious if you have considered syncing via Dropbox / Drive / OneDrive etc.

**Comment by saket** [comment]
@okmanideep I made this decision a few months ago so I'm a bit hazy on the details. Let me try recalling them:

- I realized that adding support for just one service won't be enough. People have different preferences for Dropbox, Drive, OneDrive, etc. I'd have to add support for atleast the two major ones -- Dropbox and Drive. This felt like a lot more work.

- I felt overwhelmed and lost in the APIs of both Dropbox and Drive. 

- I really like the simplicity of using git and believe that users in my demography will feel the same.

- Revision history of both Dropbox and Drive are very limited. Git is unlimited.
